### ISSUE TITLE
Executing a Singularity Test fails

### ISSUE THREAD

**Issue by sylabskeith-zz** [apptainer/singularity#1304]
## Version of Singularity:

2.4.2

### Expected behavior

BootStrap: docker
From: alpine:latest

%test
    [[ -z "$THIS_IS_A_TEST" ]] && echo "Environment is not loaded." >&2
    echo "test: THIS_IS_A_TEST=$THIS_IS_A_TEST"

%runscript
    [[ -z "$THIS_IS_A_TEST" ]] && echo "Environment is not loaded." >&2
    echo "runscript: THIS_IS_A_TEST=$THIS_IS_A_TEST"

%post
    apk update
    apk add bash
    echo 'export THIS_IS_A_TEST=TRUE' >> /environment

###Actual Behavior

1.	The environment is not loaded before test during bootstrap
2.	Singularity test command is failing with this msg:  "No Singularity tests for contained app:"

**Comment by dtrudg** [comment]
Hi @sylabskeith 

2. is fixed by the #1335 that was merged into the 2.4.3 release:
```
$ singularity test test.img
test: THIS_IS_A_TEST=TRUE
```

1. is not - the environment is not loaded for test during build:
```
Running test scriptlet
+ [[ -z  ]]
+ echo Environment is not loaded.
Environment is not loaded.
+ echo test: THIS_IS_A_TEST=
test: THIS_IS_A_TEST=
```

This is by design from a change implemented in #1071 where the rationale was discussed in #1053 - environment is no longer sourced during the build.

**Comment by dtrudg** [comment]
@sylabskeith - closing this as the fix for behaviour (2) has been merged, and  (1) is by design, as above. Let us know if any further issue. Thanks.
